srvutil/metrics.go

```diff
@@ -40,14 +52,27 @@ func (w *httpRecorder) LogFields() logrus.Fields {
 	return nil
 }
 
+// Hijack implements the http.Hijacker interface, to allow for e.g. WebSockets.
+func (w *hijackableRecorder) Hijack() (net.Conn, *bufio.ReadWriter, error) {
+	return w.Hijacker.Hijack()
+}
+
+func newHTTPRecorder(w http.ResponseWriter) loggableHTTPRecorder {
+	recorder := &httpRecorder{ResponseWriter: w}
+	if hijacker, ok := w.(http.Hijacker); ok {
+		return &hijackableRecorder{recorder, hijacker}
```

Comment:
Am I crazy or should
```go
type hijackableRecorder struct {
	httpRecorder
}
```
be
```go
type hijackableRecorder httpRecorder
```
?

Also, I think your implementation would still need `if _, ok := w.(http.Hijacker); ok` in `(w *hijackableRecorder) Hijack()`, with a panic on `!ok`.

Reply:
It would panic with a proper error anyway, so no harm no foul.

And I did try your suggestion previously, it would work, but we’d have to reimplement all methods and reconvert+delegate the methods calls to `httpRecorder`, it doesn't automatically inherit the methods.

Reply:
Ah, makes sense.